lib/location/method_gnss.c

```diff
@@ -816,6 +819,17 @@ static void method_gnss_positioning_work_fn(struct k_work *work)
 	location_core_timer_start(gnss_config.timeout);
 }
 
+#if defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_NRF_CLOUD_PGPS)
+static void method_gnss_agps_req_work_fn(struct k_work *work)
+{
+	/* Start and stop GNSS to trigger NRF_MODEM_GNSS_EVT_AGPS_REQ event if assistance data
+	 * is needed.
+	 */
+	nrf_modem_gnss_start();
+	nrf_modem_gnss_stop();
```

Comment:
not related to this PR as this code was run previously in a caller thread, but  looks strange. I was wondering why is this kind of start/stop really needed? We are doing start() later anyways, why that isn't enough for triggering NRF_MODEM_GNSS_EVT_AGPS_REQ?

Reply:
The reason for this is that we want to get the `NRF_MODEM_GNSS_EVT_AGPS_REQ` immediately to be able to fetch A-GPS data if needed, but don't want to start GNSS before the RRC connection has been released.

It's possible to ask for A-GPS data need also using GNSS API function `nrf_modem_gnss_agps_expiry_get()`, but there's a problem related to filtered ephemerides because of which we haven't taken it into use yet.

Reply:
ok, thanks for the clarifications. We can consider improvements on this but that would be a topic of future PRs.